internal/controller/suite_util_test.go

```diff
@@ -649,6 +721,54 @@ func waitServiceAccount(serviceAccountKey types.NamespacedName) corev1.ServiceAc
 	return serviceAccount
 }
 
+func waitServiceAccountLinkedSecret(serviceAccountKey types.NamespacedName, secretName string, isPullSecret bool) corev1.ServiceAccount {
+	serviceAccount := corev1.ServiceAccount{}
+	Eventually(func() bool {
+		if err := k8sClient.Get(ctx, serviceAccountKey, &serviceAccount); err != nil {
+			return false
+		}
+		if isPullSecret {
+			for _, pullSecretReference := range serviceAccount.ImagePullSecrets {
+				if pullSecretReference.Name == secretName {
+					return true
+				}
+			}
+		} else {
+			for _, secretReference := range serviceAccount.Secrets {
+				if secretReference.Name == secretName {
+					return true
+				}
+			}
+		}
+		return false
+	}, timeout, interval).Should(BeTrue())
+
+	return serviceAccount
+}
+
+func waitServiceAccountLinkedSecretGone(serviceAccountKey types.NamespacedName, secretName string, isPullSecret bool) {
+	serviceAccount := corev1.ServiceAccount{}
+	Eventually(func() bool {
+		if err := k8sClient.Get(ctx, serviceAccountKey, &serviceAccount); err != nil {
+			return false
+		}
+		if isPullSecret {
+			for _, pullSecretReference := range serviceAccount.ImagePullSecrets {
+				if pullSecretReference.Name == secretName {
+					return false
+				}
+			}
+		} else {
+			for _, secretReference := range serviceAccount.Secrets {
+				if secretReference.Name == secretName {
+					return false
+				}
+			}
+		}
+		return true
+	}, timeout, interval).Should(BeTrue())
```

Comment:
Could this be simplified with `isSaSecretLinked`?

Reply:
Technically yes, but I think we shouldn't use main code in tests, because tests should test.

Reply:
Ok, fair point